### DISCUSSION TITLE
Why can't I have two versions at once? I thought path was for it?

### DISCUSSION THREAD

**Discussion by Milo-W** [coreybutler/nvm-windows#1155] [General] [answered]
Installed two versions of node.js
two separate cmd windows, set two separate root paths
when I choose one in one window version and the other window a different version, it's all the same for both.

Why?
This is idiotic.

**Comment by coreybutler** [comment, marked as answer]
The different use cases are spelled out in the Node version management repo. There are millions of users leveraging system-wide Node.js installations. Remember, this is how Node.js was intended to operate, before version managers were created.

While there are per-shell use cases, they are part of a larger environment management solution (not just version management). Some tools like nvm, n, and volta dabble in this. I feel they are incomplete solutions, which often create large vulnerability surface areas. For example, in corporate environments, it's common to see older insecure runtimes exploited simply because they are accessible to other programs on the same system (i.e. vulnerable app calls `node virus.js` with an nvmrc file specifying an older version). A system-wide instance can be vulnerable too, but upgrading at least reduces the attack surface. This is why several military, government, and financial organizations use NVM4W. Again, I feel the state of "environment" management is very lacking in version management tools. This is why we've been building a tool (Runtime) specifically for that purpose, and explicitly separated from this project.

That's all the detail I'm going into. I don't take kindly to being called idiotic, so I'd recommend finding another tool/community to serve your needs.
